veps/sig-storage/dic-on-heterogeneous-cluster/dic-on-heterogeneous-cluster.md

```diff
@@ -0,0 +1,538 @@
+
+#### Create Multiple `DataImportCronTemplate` Objects, One for Each Architecture
+HCO could create a new `DataImportCronTemplate` object for each architecture in the `SSP` CR, and the user will be
+responsible for doing the same for custom `DataImportCronTemplate` objects.
+
+pros:
+
+- no changes are required from SSP, except for adopting the new CDI API, and changes regarding the VM templates.
+
+cons:
+
+- the user will be responsible for creating and maintaining the `DataImportCronTemplate` objects, and
+  the user will need to create a new `DataImportCronTemplate` object for each architecture.
+- the `SSP` CR will be huge, and will contain a lot of `DataImportCronTemplate` objects. It will be hard to maintain it
+  or understand it.
+- it looses the meaning of the templating mechanism, by the `DataImportCronTemplate` type.
+
+#### Create Multiple `DataSource` CRs from a Single `DataImportCron` Object
+No changes in SSP, regarding the `DataImportCronTemplate` objects. SSP will create a single `DataImportCron` CR for each
+`DataImportCronTemplate` object in the `SSP` CR, as done today. The `ssp.kubevirt.io/dict.architectures` (with a new
+name, e.g. `cdi.kubevirt.io/dict.architectures`) annotation will be copied to the `DataImportCron` CR, as any other
+annotation.
+
+CDI will create multiple `DataSource` CRs, one for each architecture, according to the
+`cdi.kubevirt.io/dict.architectures` annotation, and will import the images into architecture specific `VolumeSnapshot`
+resources. 
+
+When performing the actual image import, if the `cdi.kubevirt.io/dict.architectures` annotation is set in
+the `DataImportCron` CR, the CDI will create architecture specific `DataVolume` CR, for each architecture
+listed in the annotation. CDI will set the `DataVolome` CR's `spec.source.registry.platform.architecture` field, to the
+required architecture.
+
+CDI will add the `template.kubevirt.io/architecture` label to the `DataSource` CR, with the architecture of the image,
+and will set the `cdi.kubevirt.io/storage.import.datasource-name` label to the value of the `spec.managedDataSource` 
+field in the `DataImportCron` CR.
+
+CDI will need to remove any orphan `DataSource` or `VolumeSnapshot` CRs, that are not referenced by any `DataImportCron`
+CR.
+
+SSP will still need to create architecture specific templates, and will still need to set the `DataSource` name 
+in each template, to the architecture specific `DataSource` CR name.
+
+Open questions:
+1. SSP currently creates the `DataSource` CRs in some cases. We have two options here:
+   1. SSP will have to have the same logic as in CDI, when creating the architecture specific `DataSource` CRs.
+   2. SSP will no longer create the `DataSource` CRs, and will let CDI to do that.
+
+2. What is the best way for SSP to find out which architecture specific templates to create? 
+   1. SSP can use the new `spec.cluster.workloadArchitectures` field in the `SSP` CR, and create the templates
```

Comment:
We have other Templates without a default DataSource (Windows), to me that's acceptable. For most templates we provide we should have a fitting disk image anyway.

Reply:
That leaves it open to a user to provide their own image. Keep in mind that DataSources in Templates are not hardcoded, they can be provided as a parameter.

Reply:
Btw the default DataSurces for windows exists (they are created by SSP), we just don't provide images for them (it is user responsibility).
The default `DATA_SOURCE_NAME` is hardcoded in templates (e.g. https://github.com/kubevirt/common-templates/blob/master/templates/windows11.tpl.yaml#L202), it is not set by ssp operator.

Reply:
> The default DATA_SOURCE_NAME parameter is hardcoded in templates (e.g. https://github.com/kubevirt/common-templates/blob/master/templates/windows11.tpl.yaml#L202), it is not set by ssp operator.

This is one of the thing we need to change as part of the implementation of this VET, as you mentioned at https://github.com/kubevirt/common-templates/issues/680. And it's also detailed in the VET, in the "Creating a Virtual Machine Templates" section.